### PR TITLE
Add MMCV Faster-RCNN training example and doc

### PR DESCRIPTION
**Pull request by PatrickkZ**
## Description

- Add a mmcv faster-rcnn training example with orca
- Add readme
### 1. Why the change?

related issue: https://github.com/intel-analytics/BigDL/issues/7774
### 2. User API changes
N/A

### 3. Summary of the change 

Add a mmcv faster-rcnn training example with orca and related doc.
### 4. How to test?
- [ ] N/A
- [x] Unit test
- [x] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+mkdir checkpoints
+wget https://download.openmmlab.com/mmdetection/v2.0/faster_rcnn/faster_rcnn_r50_caffe_fpn_mstrain_3x_coco/faster_rcnn_r50_caffe_fpn_mstrain_3x_coco_20210526_095054-1f77628b.pth
+```
+
+## Running commands
+
+- spark local mode
+
+```bash
+python train.py  --dataset /your/dataset/path/to/kitti_tiny/ --config $MMDET_PATH/configs/faster_rcnn/faster_rcnn_r50_caffe_fpn_mstrain_1x_coco.py --load_from $MMDET_PATH/checkpoints/faster_rcnn_r50_caffe_fpn_mstrain_3x_coco_20210526_095054-1f77628b.pth
```
Comment (leonardozcm) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:47]: Add descriptions on these args

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+`$MMDET_PATH`: your mmdetection repo local path.
+
+The result should looks like this.
```
Comment (leonardozcm) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:52]: no should

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+(MMCVRayEpochRunner pid=188889) 2023-03-13 10:48:45,976 - mmdet - INFO - Epoch [1][25/25]       lr: 2.500e-03, eta: -1 day, 23:59:54, time: 5.530, data_time: 0.092, loss_rpn_cls: 0.0237, loss_rpn_bbox: 0.0160, loss_cls: 0.3650, acc: 87.1133, loss_bbox: 0.3418, loss: 0.7465, AP50: 0.5480, mAP: 0.5476
+Stopping orca context
+```
+
+- yarn-client mode
+
+If you want to use the pre-trained checkpoints, please first upload the checkpoint file to HDFS, then execute
+
+```bash
+python train.py  --dataset /your/dataset/path/to/kitti_tiny/ --cluster_mode yarn-client --config $MMDET_PATH/configs/faster_rcnn/faster_rcnn_r50_caffe_fpn_mstrain_1x_coco.py --load_from hdfs://ip:port/your/hdfs/path/to/faster_rcnn_r50_caffe_fpn_mstrain_3x_coco_20210526_095054-1f77628b.pth
```
Comment (leonardozcm) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:168]: Do we need to upload dataset to hdfs or pre-download it on each node?

**Comment by leonardozcm** [comment]
@hkvision please take a look?

File: python/orca/dev/example/run-example-test-ray.sh [hunk 4]
```diff
@@ -207,3 +237,4 @@ echo "#11 auto-xgboost-regressor-spark-df example time used:$time12 seconds"
 echo "#12 ray-dataset-xgboost example time used:$time13 seconds"
 echo "#13 orca brainMRI example time used:$time14 seconds"
 echo "#14 orca resnet50 inference example time used:$time15 seconds"
+echo "#15 mmcv faster_rcnn training example time used:$time16 seconds"
```
Comment (hamham223) [inline python/orca/dev/example/run-example-test-ray.sh:240]: Could you plz rebase the pr? I made some change to the test script last week. Align the numberings to make them beautiful! 😄 

Previous PR: https://github.com/intel-analytics/BigDL/pull/7851/files

File: python/orca/example/learn/mmcv/faster_rcnn/README.md (hunk 3 above)
Comment (PatrickkZ) [reply python/orca/example/learn/mmcv/faster_rcnn/README.md:168]: yes, I will add more explanation to clarify this.

File: python/orca/dev/example/run-example-test-ray.sh (hunk 4 above)
Comment (PatrickkZ) [reply python/orca/dev/example/run-example-test-ray.sh:240]: sure

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+## Prepare the dataset
+
+This training example use the kitti_tiny dataset, you can download it from [here](https://download.openmmlab.com/mmdetection/data/kitti_tiny.zip) and unzip the dataset.
+
+## Prepare the environment
+
+We recommend you to use Anaconda to prepare the environment:
+
+```
+conda create -n bigdl-orca pyton=3.7  # "bigdl-orca" is conda environment name, you can use any name you like.
```
Comment (hkvision) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:12]: python typo

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+conda activate bigdl-orca
+
+pip install torch torchvision tensorboard
+pip install -U openmim
+mim install mmcv-full
+pip install mmdet
+
+pip install --pre --upgrade bigdl-orca
+pip install ray
```
Comment (hkvision) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:21]: use bigdl-orca[ray]

File: python/orca/dev/example/run-example-test-ray.sh (hunk 4 above)
Comment (hkvision) [reply python/orca/dev/example/run-example-test-ray.sh:240]: Contact @hamham223 to run this for the nightly build test; example test is now not included in the PR test to save time.

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+In this example, we need to use the config file from the [mmdet repo](https://github.com/open-mmlab/mmdetection).
+
+```
+git clone https://github.com/open-mmlab/mmdetection
+```
+
+The config file we use in this example is `mmdetection/configs/faster_rcnn/faster_rcnn_r50_caffe_fpn_mstrain_1x_coco.py`
+
+If you want to use the pre-trained checkpoints for further training, also download the checkpoint file
```
Comment (hkvision) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:33]: add `:` at the end

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+```
+cd mmdetection
+mkdir checkpoints
+wget https://download.openmmlab.com/mmdetection/v2.0/faster_rcnn/faster_rcnn_r50_caffe_fpn_mstrain_3x_coco/faster_rcnn_r50_caffe_fpn_mstrain_3x_coco_20210526_095054-1f77628b.pth
+```
+
+## Running commands
+
+- spark local mode
```
Comment (hkvision) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:43]: keep this format the same as other examples? e.g. add arguments explanation; put the results at the very end

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+```python
+conda create -n bigdl-orca pyton=3.7  # "bigdl-orca" is conda environment name, you can use any name you like.
+conda activate bigdl-orca
+
+pip install torch torchvision tensorboard
+pip install -U openmim
+mim install mmcv-full
+pip install mmdet
+
+pip install --pre --upgrade bigdl-orca[ray]
```
Comment (leonardozcm) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:20]: why these two lines looks strange?

File: python/orca/example/learn/mmcv/faster_rcnn/README.md [hunk 10]
```diff
@@ -0,0 +1,191 @@
+# MMCV Faster-RCNN Training Example
+
+## Prepare the dataset
+
+This training example use the kitti_tiny dataset, you can download it from [here](https://download.openmmlab.com/mmdetection/data/kitti_tiny.zip) and unzip the dataset.
```
Comment (leonardozcm) [inline python/orca/example/learn/mmcv/faster_rcnn/README.md:5]: uses

File: python/orca/example/learn/mmcv/faster_rcnn/README.md (hunk 9 above)
Comment (PatrickkZ) [reply python/orca/example/learn/mmcv/faster_rcnn/README.md:20]: In the actual doc, this looks normal.

File: python/orca/example/learn/mmcv/faster_rcnn/train.py [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+                    else:
+                        gt_labels_ignore.append(-1)
+                        gt_bboxes_ignore.append(bbox)
+
+                data_anno = dict(
+                    bboxes=np.array(gt_bboxes, dtype=np.float32).reshape(-1, 4),
+                    labels=np.array(gt_labels, dtype=np.long),
+                    bboxes_ignore=np.array(gt_bboxes_ignore,
+                                           dtype=np.float32).reshape(-1, 4),
+                    labels_ignore=np.array(gt_labels_ignore, dtype=np.long))
```
Comment (hamham223) [inline python/orca/example/learn/mmcv/faster_rcnn/train.py:88]: This is removed after numpy 1.24, just use `int`

Referene: https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated

**Comment by PatrickkZ** [comment]
nightly test pass：

image

**Comment by leonardozcm** [comment]
move to https://github.com/intel-analytics/BigDL/pull/7956
